README.md

```diff
@@ -1,2 +1,278 @@
 # sailfish-connman-plugin-iptables
-Connman plugin that provides DBus API for controlling iptables rules
+
+Connman plugin that provides D-Bus API for controlling iptables rules.
+
+## API description
+
+The detailed API description is provided as D-Bus introspect XML file (in
+spec/sailfish_iptables_dbus_interface_description.xml) from which a docbook XML
+is generated for the docs-package (run make -C doc). 
+
+File contains descriptions for all the methods and parameters for each method.
+
+### Generating documentation HTML
+
+A html page can be generated from the introspect XML with the script at doc/docbook_html. The script requires xsltproc, docbook, docbook-xsl, docbook-xsl-ns, and docbook5-xml in order to generate the file from docbook XML.
+
+## Loading the plugin
+
+Plugin is loaded by Connection Manager at startup.
+
+When loaded plugin registers its interface "org.sailfishos.connman.mdm.iptables"
+under "net.connman" with path "/org/sailfishos/connman/mdm/iptables". Plugin
+uses Connection Manager D-Bus functions to register itself to D-Bus.
+
+After the plugin is loaded signal "Initialize" is sent over D-Bus indicating
+that the plugin is running. Then iptables filter table content is loaded from
+the default save path of connman iptables module for IPv4 rules. All previously
+created custom chains (prefixed with sfos_) are loaded at plugin startup (using
+iptables content returned by connman) into plugin's internal database.
+
+When unloaded (by Connman at exit) iptables filter table is saved to connman's
+default save path, iptables content is cleared, custom chains are removed from
+the filter table and finally signal "Shutdown" is sent over D-Bus. Custom chains
```

Comment:
This `Shutdown` signal seems to be useless (unlike `Initialize`). connman never unloads plugins. If the client needs to know when connman disappears from the bus, it would watch `net.connman` name - and be notified even if connman has crashed (in which case `Shutdown` signal won't be sent). I don't see any use case for this `Shutdown` thing.

Reply:
I find this necessary (both Shutdown and Initialize) as: the plugin may not be loaded because of version mismatch (highly unlikely but still possible), the plugin is registered/de-registered some time after connman is started or stopped that it is easier to follow the state than to send a message and get a reply telling that service does not exist and since this is the API 3rd parties are used who are not interested or should be using ConnMan directly they should have one place to listen to status changes.

Reply:
I wasn't complaining about `Initialize`! `Initialize` is ok. My point was that listening for `Shutdown` is (currently) equivalent to watching the `net.connman` D-Bus name, except that watching the name is significantly more reliable. Whether or not 3rd parties are interested in using ConnMan directly, they still need this name, otherwise how would they find the service providing this interface.

Reply:
But if you like it, leave it there! There's no harm in it.